astropy/coordinates/representation.py

```diff
@@ -171,6 +171,19 @@ def __init__(self, *args, **kwargs):
         for component, attr in zip(components, attrs):
             setattr(self, '_' + component, attr)
 
+    def __eq__(self, other):
+        if not isinstance(other, BaseRepresentationOrDifferential):
+            return False
+
+        if isinstance(other, self.__class__):
+            for comp in self.components:
+                if np.any(getattr(self, comp) != getattr(other, comp)):
+                    return False
```

Comment:
Why are we not returning the array here?

Reply:
Ah, very good point. I had originally been thinking that there's no natural way to do it due to components being ambiguous, but now that you bring it up you're absolutely right that it's just an `np.all(...)` outside the loop.